### PR TITLE
Feature/builder pickup

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Remove duplicated guard code

Review please

### CODE REVIEW

**Comment by Raycoms** [comment]
@Kostronor

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java [hunk 1]
```diff
@@ -28,19 +27,16 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraftforge.items.wrapper.InvWrapper;
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java]: MINOR Remove this unused import 'org.jetbrains.annotations.Nullable'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 39 issues
* ![MINOR][MINOR] 39 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [jobs](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/colony/jobs): Add a 'package-info.java' file to document the 'jobs' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 38 issues
* ![MINOR][MINOR] 38 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [jobs](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/colony/jobs): Add a 'package-info.java' file to document the 'jobs' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/704f2e3565a44accf40ab32559137d917bfe3c07/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 42 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 38 minor
* ![INFO][INFO] 2 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L59](https://github.com/Minecolonies/minecolonies/blob/6852dd7fa9576c257c6cbcd0d9c0a46d45f97399/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L59): class "AbstractEntityAIStructure" has 37 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIStructure.java#L777](https://github.com/Minecolonies/minecolonies/blob/6852dd7fa9576c257c6cbcd0d9c0a46d45f97399/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L777): Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/6852dd7fa9576c257c6cbcd0d9c0a46d45f97399/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/6852dd7fa9576c257c6cbcd0d9c0a46d45f97399/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/6852dd7fa9576c257c6cbcd0d9c0a46d45f97399/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/6852dd7fa9576c257c6cbcd0d9c0a46d45f97399/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/6852dd7fa9576c257c6cbcd0d9c0a46d45f97399/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/6852dd7fa9576c257c6cbcd0d9c0a46d45f97399/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/6852dd7fa9576c257c6cbcd0d9c0a46d45f97399/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/6852dd7fa9576c257c6cbcd0d9c0a46d45f97399/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marvin-bitterlich** [comment]
:shipit: but having #1000 with the addition would be nice

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1002/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**